src/main/java/com/shapesecurity/functional/data/ImmutableSet.java

```diff
@@ -72,4 +101,11 @@ public boolean contains(@Nonnull T datum) {
     public boolean equals(Object other) {
         return other instanceof ImmutableSet && this.data.length == ((ImmutableSet) other).data.length && this.data.foldLeft((memo, pair) -> memo && ((ImmutableSet) other).data.containsKey(pair.left), true);
     }
+
+    @Override
+    public Iterator<T> iterator() {
+        return this.toList().iterator();
```

Comment:
We can do this without having to do a full copy, though it might require making HashTable iterable (presumably of `Pair<K, V>`), which seems reasonable enough. Take a look at the [ConcatList iterator](https://github.com/shapesecurity/shape-functional-java/blob/48b68fd0ce54d7e20142f2a367dbb3754a62a6ef/src/main/java/com/shapesecurity/functional/data/ConcatList.java#L505); we'd use a similar strategy.